src/pages/VenuePage/VenuePage.tsx

```diff
@@ -93,57 +94,49 @@ const VenuePage: React.FC = () => {
 
   // NOTE: User location updates
 
-  const updateUserLocationToCurrentVenue = useCallback(() => {
-    if (!userId || !venueName) return;
-
-    trackLocationEntered({ userId, locationName: venueName });
-  }, [userId, venueName]);
-
   useInterval(() => {
-    updateUserLocationToCurrentVenue();
+    if (!userId || !profile?.lastSeenIn) return;
+
+    updateCurrentLocationData({
+      userId,
+      profileLocationData: profile.lastSeenIn,
+    });
```

Comment:
What will happen if they don't have `profile.lastSeenIn` (is that even possible?). Won't that make this never run? Or is this more about `profile` not being present? In which case, would we be better to just make it `!profile` instead?

Since `updateCurrentLocationData` is only taking the first entry from `profileLocationData` anyway, and then updating the time on that key, it feels weird that we need to pass the full thing through. Wouldn't it just be simpler to pass the name of the key we want to update through directly, and reduce all of this additional complexity?

Or alternatively, even though I believe you said that there can/should only ever be a single entry in `lastSeenIn`, since nothing actually enforces that currently, would it not be more robust to code `updateCurrentLocationData` to update every location key that is present? Or something to that effect?

Reply:
if profile.lastSeenIn is empty, there is nothing to update, thus it will not update. We do set locations in a bunch of places. I don't know the way it is going to evolve, so I'm not sure it is efficient to think that ahead. If there is change in logic in the future, a changer should understand the way it works.

profileLocationData has only one key inside. I could pass this key, but why? What if we decide to change the logic in the future?

Reply:
> profileLocationData has only one key inside. I could pass this key, but why? What if we decide to change the logic in the future?

My desire was to align what is passed into the function, with what the function is doing. I don't feel particularly strongly about changing the parameter, your reasoning seems to make sense RE: continuing to use `LocationData`

> I don't know the way it is going to evolve, so I'm not sure it is efficient to think that ahead. If there is change in logic in the future, a changer should understand the way it works.

Fair enough. I guess if you haven't already, can you add a comment or two in `updateCurrentLocationData` explaining why it's only taking the first param/updating that, not the entire collection/etc. That would satisfy my request here.